src/assets.rs

```diff
@@ -324,40 +517,52 @@ impl Assets {
         definition: AssetDefinition,
         mint_info: Option<MintInfo>,
     ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
-        let mut asset = Asset {
+        let asset = Asset {
             definition,
             name: None,
             description: None,
             icon: None,
             mint_info,
             verified: false,
         };
-        // The asset is verified if it's in the verified set.
-        if self.verified_assets.contains(&asset.definition.code) {
-            asset.verified = true
-        }
-        let store_asset = self.store.load(&asset.definition.code);
-        let mut editor = AssetEditor::new(&mut self.store, asset);
-        if let Ok(store_asset) = store_asset {
-            editor.update(store_asset);
-        }
-        editor.save::<L>()?;
-        Ok(editor)
+        self.insert(asset)
     }
 
     /// Create a native asset.
     ///
     /// Returns the editor for the created asset.
-    fn create_native<L: Ledger>(&mut self) -> Result<AssetEditor<'_>, KeystoreError<L>> {
-        let mut asset = Asset::native::<L>();
-        // The asset is verified if it's in the verified set.
-        if self.verified_assets.contains(&asset.definition.code) {
-            asset.verified = true
-        }
+    pub fn create_native<L: Ledger>(&mut self) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        let asset = Asset::native::<L>();
+        self.insert(asset)
+    }
+
+    /// Create an asset internally.
+    ///
+    /// If the store doesn't have an asset with the same code, adds the created asset to the store.
+    /// Otherwise, updates the exisiting asset.
+    ///
+    /// Returns the editor for the created asset.
+    pub(crate) fn create_internal<L: Ledger>(
```

Comment:
The `*_internal` functions are very similar to the non-internal functions except that the given asset takes precedence over the existing asset in the store when updating metadata (name, description, etc.) only if the given asset is verified, whereas in the non-internal versions, the given asset can be unverified as long as the existing asset is also unverified.

This prevents users from accidentally overwriting an existing asset with an unverified asset when importing. For other scenarios where a user indeed wants to update an asset with an unverified asset, the non-internal functions should be used.

I guess part of the reason this caused confusion was because of the function names. I haven't come up with a better name than `internal`.